File: app/src/main/java/pl/rpieja/flat/dto/Summary.java
package pl.rpieja.flat.dto;

import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * Created by radix on 2017-11-01.
 */

public class Summary implements ChargeLike {
    public double amount;
    public String name;
    public int room, id;

    @Nullable
    @Override
    public Double getChargeAmount() {
        return amount;
    }

    @Nullable
    @Override
    public String getChargeName() {
        return name;
    }

    @Nullable
    @Override
    public List<User> getFromUsers() {
        return null;
    }

    @Nullable
    @Override
    public List<User> getToUsers() {
        return null;
    }
}

File: app/src/main/java/pl/rpieja/flat/dto/Summary.kt
package pl.rpieja.flat.dto

import pl.memleak.flat.ChargesQuery

data class Summary(
        var user: User,
        var amount: Double
) : ChargeLike {
    constructor(obj: ChargesQuery.Monthly) : this(
            user = User(obj.user()!!),
            amount = obj.amount()!!
    )

    override val chargeAmount: Double
        get() = amount
    override val chargeName: String
        get() = user.name
    override val fromUsers: List<User>
        get() = emptyList()
    override val toUsers: List<User>
        get() = emptyList()
}
